weblate/api/views.py

```diff
@@ -314,6 +316,65 @@ def groups(self, request, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+    @action(
+        detail=True, methods=["get", "post"], serializer_class=NotificationSerializer
+    )
+    def notifications(self, request, **kwargs):
+        obj = self.get_object()
+        if request.method == "POST":
+            self.perm_check(request)
+            with transaction.atomic():
+                serializer = NotificationSerializer(
+                    data=request.data, context={"request": request}
+                )
+                serializer.is_valid(raise_exception=True)
+                serializer.save(user=obj)
+                return Response(serializer.data, status=status.HTTP_201_CREATED,)
+
+        queryset = obj.subscription_set.order_by("id")
+        page = self.paginate_queryset(queryset)
+        serializer = NotificationSerializer(
+            page, many=True, context={"request": request}
+        )
+
+        return self.get_paginated_response(serializer.data)
+
+    @action(
+        detail=True,
+        methods=["get", "put", "patch", "delete"],
+        url_path="notifications/(?P<subscription_id>[^/.]+)",
+        serializer_class=NotificationSerializer,
+    )
+    def notifications_details(self, request, username, subscription_id):
+        obj = self.get_object()
+
+        try:
+            subscription = obj.subscription_set.get(id=subscription_id)
+        except (Subscription.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        if request.method == "DELETE":
+            self.perm_check(request)
+            subscription.delete()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        if request.method == "GET":
+            serializer = NotificationSerializer(
+                subscription, context={"request": request}
+            )
+        else:
+            self.perm_check(request)
+            serializer = NotificationSerializer(
+                subscription, data=request.data, context={"request": request},
```

Comment:
I guess patch will not work properly as this is not handling partial data.

Reply:
Ah yeah right. I need to do the patch check and handle partial data in serializer and write tests for that.

Reply:
Added a commit for the same. Should handle it.